api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -756,110 +760,70 @@ public List<Patient> findPatients(String query, boolean includeVoided, Integer s
-		if (attributesSize > tmpStart) {
-			ListPart<Object[]> personAttributes = attributeQuery.listPartProjection(tmpStart, tmpLength, "person.personId");
-			personAttributes.getList().forEach(personAttribute -> patients.add(getPatient((Integer) personAttribute[0])));
-		}
+		patients = SearchQueryUnique.search(searchSessionFactory,
+			SearchQueryUnique.newQuery(PatientIdentifier.class, f -> 
+					newPatientIdentifierSearchPredicate(f, query, includeVoided, false), 
+				 "patient.personId", PatientIdentifier::getPatient).join(
+					 SearchQueryUnique.newQuery(PersonName.class, f -> 
+					personQuery.getPatientNameQuery(f, query, includeVoided),
+					 "person.personId", pN -> getPatient(pN.getPerson().getId())).join(
+						 SearchQueryUnique.newQuery(PersonAttribute.class,
+						f -> personQuery.getPatientAttributeQuery(f, query, includeVoided), 
+						"person.personId", pA -> getPatient(pA.getPerson().getId())
+						))), start, length);
 
 		return patients;
 	}
-	private LuceneQuery<PatientIdentifier> getPatientIdentifierLuceneQuery(String query, List<PatientIdentifierType> identifierTypes, boolean matchExactly) {
-		LuceneQuery<PatientIdentifier> patientIdentifierLuceneQuery = getPatientIdentifierLuceneQuery(query, matchExactly);
-		List<Integer> identifierTypeIds = new ArrayList<Integer>();
-		for(PatientIdentifierType identifierType : identifierTypes) {
-			identifierTypeIds.add(identifierType.getId());
-		}
-		patientIdentifierLuceneQuery.include("identifierType.patientIdentifierTypeId", identifierTypeIds);
-		patientIdentifierLuceneQuery.include("patient.isPatient", true);
-		patientIdentifierLuceneQuery.skipSame("patient.personId");
-		
-		return patientIdentifierLuceneQuery;
-	}
 	
-	private LuceneQuery<PatientIdentifier> getPatientIdentifierLuceneQuery(String paramQuery, boolean matchExactly) {
-		String query = removeIdentifierPadding(paramQuery);
-		List<String> tokens = tokenizeIdentifierQuery(query);
-		query = StringUtils.join(tokens, " OR ");
-		List<String> fields = new ArrayList<>();
-		fields.add("identifierPhrase");
-		fields.add("identifierType");
-		String matchMode = Context.getAdministrationService()
-			.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_IDENTIFIER_SEARCH_MATCH_MODE);
-		if (matchExactly) {
-			fields.add("identifierExact");
-		}
-		else if (OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_SEARCH_MATCH_START.equals(matchMode)) {
-			fields.add("identifierStart");
-		} 
-		else  {
-			fields.add("identifierAnywhere");
-		}
-		return LuceneQuery.newQuery(PatientIdentifier.class, sessionFactory.getCurrentSession(), query, fields);
+	private SearchPredicate getPatientIdentifierSearchPredicate(SearchPredicateFactory f, String paramQuery, boolean matchExactly) {
+		List<String> tokens = tokenizeIdentifierQuery(removeIdentifierPadding(paramQuery));
+		final String query = StringUtils.join(tokens, " | ");
+		//TODO: hibernate search identifierType?
+		//fields.add("identifierType");
```

Comment:
The implementation didn't make sense to me... It was searching by identifierType DB id, which was probably not what the intention was. The search should specify identifierType.name for example. No test for that though....